.github/workflows/deploy-gcp-tests.yml

```diff
@@ -196,24 +196,42 @@ jobs:
           service_account: 'github-service-account@zealous-zebra.iam.gserviceaccount.com'
           token_format: 'access_token'
 
-      # Before executing any further steps, validate the local state and remote version are the same,
-      # or at least that the local state version is greater than the available cached state version from main.
+      # Find a cached state disk for this job, matching all of:
+      # - disk kind (disk_prefix)
+      # - disk target height (disk_suffix)
+      # - state version (from the source code)
       #
-      # Aftwards, get the disk name to be used on further steps
-      - name: Validate local state version with cached state
+      # If there are multiple disks:
+      # - prefer images generated from the `main` branch, then any other branch
+      # - prefer newer images to older images
+      #
+      # Passes the disk name to subsequent steps using an environmental variable.
+      - name: Find cached state disk
         id: get-disk-name
         run: |
           LOCAL_STATE_VERSION=$(grep -oE "DATABASE_FORMAT_VERSION: .* [0-9]+" "$GITHUB_WORKSPACE/zebra-state/src/constants.rs" | grep -oE "[0-9]+" | tail -n1)
           echo "LOCAL_STATE_VERSION: $LOCAL_STATE_VERSION"
 
-          ZEBRA_STATE_DISK=$(gcloud compute images list --filter="name~${{ inputs.disk_prefix }} AND name~-${{ inputs.disk_suffix }}" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
-          echo "Disk: $ZEBRA_STATE_DISK"
-          echo "Description: $(gcloud compute images describe $ZEBRA_STATE_DISK --format='value(DESCRIPTION)')"
+          # Try to find an image generated from the main branch
+          # name~${{ inputs.disk_prefix }}-main- AND
+          ZEBRA_STATE_DISK=$(gcloud compute images list --filter="name~-v${LOCAL_STATE_VERSION}- AND name~-${{ inputs.disk_suffix }}" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
+          echo "main Disk: $ZEBRA_STATE_DISK"
+
+          if [[ -z "$ZEBRA_STATE_DISK" ]]; then
+              # Try to find an image generated from any other branch
+              # name~${{ inputs.disk_prefix }}- AND
+              ZEBRA_STATE_DISK=$(gcloud compute images list --filter="name~-v${LOCAL_STATE_VERSION}- AND name~-${{ inputs.disk_suffix }}" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
+              echo "Disk: $ZEBRA_STATE_DISK"
+          fi
```

Comment:
```suggestion
          # Try to find an image generated from the main branch
          # name~${{ inputs.disk_prefix }}-main- AND
          STATE_DISK=$(gcloud compute images list --filter="name~${{ inputs.disk_prefix }}-main- AND name~-v${LOCAL_STATE_VERSION}- AND name~-${{ inputs.disk_suffix }}" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
          echo "main Disk: $STATE_DISK"

          if [[ -z "$STATE_DISK" ]]; then
              # Try to find an image generated from any other branch
              # name~${{ inputs.disk_prefix }}- AND
              STATE_DISK=$(gcloud compute images list --filter="name~${{ inputs.disk_prefix }}- AND name~-v${LOCAL_STATE_VERSION}- AND name~-${{ inputs.disk_suffix }}" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
              echo "Disk: $STATE_DISK"
          fi
```

Reply:
This is a temporary change to find out why CI was failing.

This suggested change doesn't actually work - using 3 filters fails:
https://github.com/ZcashFoundation/zebra/runs/6415207684?check_suite_focus=true#step:6:229

But using two filters works:
https://github.com/ZcashFoundation/zebra/runs/6415284000?check_suite_focus=true#step:6:238

So I combined it into a single filter in the latest commit.